tensorflow/core/kernels/conv_2d_gpu.h

```diff
@@ -890,7 +899,7 @@ struct TransposeElemType<4> {
 };
 template <>
 struct TransposeElemType<8> {
-  using type = uint64;
+  using type = float2;
```

Comment:
Is this necessary for correctness? Would we run into similar issues with complex<float16> and TransposeElemType<4>?

Reply:
Yes, this is necessary. Instead of simply conjugating, we call `maybe_conj`, which is specialized for complex types in GPU (float2 and double2).

I thought about `TransposeElemType<4>` when fixing this, and came to the conclusion that tensorflow doesn't have a complex type which uses float16. `std::complex<float16>` doesn't exist, as far as I can tell.

To be honest, I'm not entirely sure why `TransposeElemType` is there in the first place. If I had to make a guess, I would say that it's in order to make this code work for non-basic types (i.e not tensorflow builtin numeric types). However, this is only used in a very specific place, so it would fail to achieve that.

Thanks for the comment, I found out I didn't fix the issue for complex128, but it's sorted now.

Reply:
Oh, right, we just have DT_COMPLEX64 and DT_COMPLEX128. Sounds OK then. If you want to get rid of TransposeElemType that's fine too (we might find a failing test).

I do think it's another good reason to have a unit test. If someone comes along and adds a smaller complex type it'd make it at least possible to find and fix this issue.